sdk/storage/azure-storage-blob/azure/storage/blob/_shared/authentication.py

```diff
@@ -76,7 +76,8 @@ def _get_verb(self, request):
     def _get_canonicalized_resource(self, request):
         uri_path = urlparse(request.http_request.url).path
         try:
-            if isinstance(request.context.transport, AioHttpTransport):
+            if isinstance(request.context.transport, AioHttpTransport) or \
+                isinstance(getattr(request.context.transport, "_transport"), AioHttpTransport):
```

Comment:
Probably we want to put a default value there, 
`getattr(request.context.transport, "_transport", default=None)`
incase request.context.transport doesn't have _transport attribute. eg. if the transport is HTTPTransport.